python/tvm/script/tir/intrin.py

```diff
@@ -51,6 +52,13 @@ def bool(imm, span):
             # nest closures so we copy the name string
             def wrap(name):
                 def f(imm, span):
+                    if name.startswith("float"):
+                        if imm == "inf":
```

Comment:
what's the type of imm? can we do `FloatImm(dtype=name, value=float(imm), span=span)` instead?

Reply:
`imm` is str. This is addressed.